src/Akka.Persistence.Sql.Hosting/SqlJournalOptions.cs

```diff
@@ -154,32 +154,41 @@ protected override StringBuilder Build(StringBuilder sb)
 
             base.Build(sb);
 
-            if (IsDefaultPlugin)
-            {
-                sb.AppendLine("akka.persistence.query.journal.sql {");
-                sb.AppendLine($"connection-string = {ConnectionString.ToHocon()}");
-                sb.AppendLine($"provider-name = {ProviderName.ToHocon()}");
+            BuildQueryConfig(sb, QueryPluginId);
+            
+            if (IsDefaultPlugin && Identifier is not "sql")
+                BuildQueryConfig(sb, "akka.persistence.query.journal.sql");
+
+            return sb;
+        }
+
+        private StringBuilder BuildQueryConfig(StringBuilder sb, string queryPluginId)
+        {
+            sb.Append(queryPluginId).AppendLine("{");
+            sb.AppendLine($"connection-string = {ConnectionString.ToHocon()}");
+            sb.AppendLine($"provider-name = {ProviderName.ToHocon()}");
                 
-                if (DatabaseOptions is not null)
-                    sb.AppendLine($"table-mapping = {DatabaseOptions.Mapping.Name().ToHocon()}");
+            if (DatabaseOptions is not null)
+                sb.AppendLine($"table-mapping = {DatabaseOptions.Mapping.Name().ToHocon()}");
 
-                if (TagStorageMode is not null)
-                    sb.AppendLine($"tag-read-mode = {TagStorageMode.ToString().ToHocon()}");
+            if (TagStorageMode is not null)
+                sb.AppendLine($"tag-read-mode = {TagStorageMode.ToString().ToHocon()}");
 
-                if (TagSeparator is not null)
-                    sb.AppendLine($"tag-separator = {TagSeparator.ToHocon()}");
+            if (TagSeparator is not null)
+                sb.AppendLine($"tag-separator = {TagSeparator.ToHocon()}");
             
-                if (ReadIsolationLevel is not null)
-                    sb.AppendLine($"read-isolation-level = {ReadIsolationLevel.ToHocon()}");
-
-                DatabaseOptions?.Build(sb);
-                
-                sb.AppendLine("}");
-            }
+            if (ReadIsolationLevel is not null)
+                sb.AppendLine($"read-isolation-level = {ReadIsolationLevel.ToHocon()}");
 
             if (QueryRefreshInterval is not null)
-                sb.AppendLine($"akka.persistence.query.journal.sql.refresh-interval = {QueryRefreshInterval.ToHocon()}");
-
+                sb.AppendLine($"refresh-interval = {QueryRefreshInterval.ToHocon()}");
+            
+            sb.AppendLine($"serializer = {Serializer.ToHocon()}");
```

Comment:
Why do we need to set a serializer explicitly in this plugin? Haven't we been actively trying to avoid doing this in all other plugins?

Reply:
We want to let the `ActorSystem` serialization settings always stay in the driver seat - back in the not-too-distant past, we have an un-overrideable dependency on Newtonsoft.Json hard-coded into Akka.Persistence that was extremely annoying to remove.